modules/extensions/src/backend/index.ts

```diff
@@ -1,9 +1,14 @@
 import * as sdk from 'botpress/sdk'
 import { asyncMiddleware as asyncMw } from 'common/http'
+import _ from 'lodash'
 
 import en from '../translations/en.json'
 import fr from '../translations/fr.json'
 
+const MIDDLEWARE_NAME = 'extensions.sendMessage'
+type Events = { [target: string]: sdk.IO.OutgoingEvent }
+const events: Events = {}
+
```

Comment:
The reason for this as described [here](https://github.com/botpress/botpress/pull/3487#discussion_r444351342) is to enable storing the dropdown outgoing options so that when the next incoming event is received we can validate and error the input accordingly. Didn't seem to make sense to have dropdown validation logic outside of the module that renders them, similarly to how single-choice node validation is done. Any recommendations as to how to fetch the options for the dropdown when handling the incoming event received AFTER the dropdown is rendered?

Reply:
Depends how the dropdown is generated (dynamic or static). If the dropdown is static, then you can load the options from the CMS anytime. 

If it's dynamic, you can store the options sent to the user in the KVS, then check the KVS when you receive a reply. Ex:

```js
await kvs.forBot(event.botId).set(`${event.target}_dropdown`, choices)
//then
const choices = await kvs.forBot(event.botId).get(`${event.target}_dropdown`)
```

In any case, that's mostly a custom implementation (or call it a "specific need"), and should not be a part of this PR (to add support for channel Twilio)

Reply:
Well, when implementing the Twilio channel (and we noticed that when using Smooch as well) is when we identified that although we are able to render the dropdown options there was no validation built-in to it like there is for the single-choice components, hence we adding this fix to that module in addition to the whitelisting of channels in there which seemed wrong. Would you guys be ok to keep it if we replace the local hashmap implementation by the KVS?

Problem with fetching from the CMS is that when the incoming response is received we have no context of the dropdown element that was rendered in the previous outbound event, hence the middlewares that were added. I suppose there is no way around that in order to store the choices sent out and fetch them again when the incoming event is received, right?

Reply:
I don't think a pull request is the ideal place to discuss this, but i'm absolutely sure there is a way to accomplish this without using a middleware. When you send the dropdown, you could pair it with an action to save the ID of the dropdown sent